File: src/dotnet/Ansight.Core/HostConnectionStatus.cs
namespace Ansight;

/// <summary>
/// Snapshot of the runtime-owned host connection state.
/// </summary>
public sealed record HostConnectionStatus(
    bool IsRuntimeActive,
    bool IsConnected,
    HostConnectionState ConnectionState,
    bool HasCachedSession,
    bool HasSavedConfig,
    bool HasBundledConfig,
    HostConnectionSummaryKind SummaryKind,
    string SummaryMessage);

File: src/ios/Sources/AnsightCore/HostConnectionStatus.swift
import Foundation

public struct HostConnectionStatus: Sendable, Codable, Equatable {
    public let isRuntimeActive: Bool
    public let isConnected: Bool
    public let connectionState: HostConnectionState
    public let hasCachedSession: Bool
    public let hasSavedConfig: Bool
    public let hasBundledConfig: Bool
    public let summaryKind: HostConnectionSummaryKind
    public let summaryMessage: String
    public let hostId: String?
    public let hostName: String?
}
